src/rules/newlineBeforeReturnRule.ts

```diff
@@ -51,6 +52,17 @@ class NewlineBeforeReturnWalker extends Lint.AbstractWalker<void> {
         return ts.forEachChild(sourceFile, cb);
     }
 
+    private getFixer(returnNode: ts.ReturnStatement, previousNode: ts.Node): Lint.Replacement {
+        const prevText = previousNode.getFullText(this.sourceFile);
+        const returnText = returnNode.getFullText(this.sourceFile);
+        const indentationOfPrevStmt = prevText.substring(0, prevText.search(/\S/)).match(/ /g)!.length;
+        return Lint.Replacement.replaceFromTo(
+                                    returnNode.getFullStart(),
+                                    returnNode.getEnd(),
+                                    `\n\n${" ".repeat(indentationOfPrevStmt) + returnText.trim()}`,
+                                );
```

Comment:
Instead of detecting any indentation, I would just insert an additional line (or two) break at `returnNode.pos`.
This probably needs some special handling for comments.
It should also use `\r\n` if that's already used in the file. See `eoflineRule.ts` for example.

Reply:
So my initial approach was:
```ts
return Lint.Replacement.appendText(returnNode.pos, NEWLINE_TYPE);
``` 
but this causes the fixer for the following test case to fail:
```ts
function foo(fn) {
    fn(); return;
          ~nil [0]
}
```

Actual `tslint` output being:
```ts
function foo(fn) {
    fn();
 return; /* leading-space here */
}
```
rather than
```ts
function foo(fn) {
    fn();
    return;
}
```

Perhaps I'm missing a simple solution?

Reply:
That's the only case that would need special handling. I don't think it's worth the effort because the code is poorly formatted before the fix, too.
The resulting code can be automatically fixed by the `align` (or eventually `indent`) rule.

Otherwise you need to worry about the correct indentation char here. Currently you assume blank only.